core/src/main/java/hudson/model/Job.java

```diff
@@ -1042,7 +1047,82 @@ public PermalinkList getPermalinks() {
         }
         return permalinks;
     }
-    
+
+    /**
+     * RSS feed for changes in this project.
+     */
+    public void doRssChangelog(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
+        class FeedItem {
+            ChangeLogSet.Entry e;
+            int idx;
+
+            public FeedItem(ChangeLogSet.Entry e, int idx) {
+                this.e = e;
+                this.idx = idx;
+            }
+
+            Run<?, ?> getBuild() {
+                return e.getParent().build;
+            }
+        }
+
+        List<FeedItem> entries = new ArrayList<FeedItem>();
+        String scmDisplayName = "";
+        if (this instanceof SCMTriggerItem) {
+            SCMTriggerItem scmItem = (SCMTriggerItem) this;
+            List<String> scmNames = new ArrayList<>();
+            for (SCM s : scmItem.getSCMs()) {
+                scmNames.add(s.getDescriptor().getDisplayName());
+            }
+            scmDisplayName = " " + Util.join(scmNames, ", ");
```

Comment:
Could be optimized to StringBuilder, issue in the migrated code anyway, so unrelated to this PR